lib/board.rb

```diff
@@ -0,0 +1,63 @@
+class Board
+
+attr_accessor :cells
+
+  #sets array w/ 9 slots & '' on each slot
+  def initialize
+      @cells =Array.new(9, " ")
+  end
+
+  #clears the @cells
+  def reset!
+    @cells =Array.new(9, " ")
+  end
+
+  #display the current cells state based on @board array
+  def display
+    puts  (" #{@cells[0]} | #{@cells[1]} | #{@cells[2]} ")
+    puts  ("-----------")
+    puts  (" #{@cells[3]} | #{@cells[4]} | #{@cells[5]} ")
+    puts  ("-----------")
+    puts  (" #{@cells[6]} | #{@cells[7]} | #{@cells[8]} ")
+  end
+
+    #display the current cells status
+  def position (position)
+      position = position.to_i
+      @cells[position-1]
+  end
+    # returns true for full board
+  def full?
+      !@cells.include?(" ")
+  end
+
+    #counts how many slots are occupied
+  def turn_count
```

Comment:
Consider using the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```